### PR TITLE
Added newRelic Adapter

### PR DESCRIPTION
**Pull request by boydbloemsma**
Closes [appwrite/appwrite#4213](https://github.com/appwrite/appwrite/issues/4209)

Data sent from the adapter (too much to screenshot):

```
(
    [timestamp] => 1666888491
    [message] => Document efgh5678 not found
    [logtype] => error
    [version] => 0.11.5
    [environment] => production
    [action] => controller.database.deleteDocument
    [namespace] => api
    [server] => digitalocean-us-001
    [user] => Array
        (
            [id] => efgh5678
        )

    [breadcrumbs] => Array
        (
            [0] => Array
                (
                    [timestamp] => 1666887991
                    [category] => http
                    [action] => DELETE /api/v1/database/abcd1234/efgh5678
                    [metadata] => Array
                        (
                            [type] => debug
                        )

                )

            [1] => Array
                (
                    [timestamp] => 1666888091
                    [category] => auth
                    [action] => Using API key
                    [metadata] => Array
                        (
                            [type] => debug
                        )

                )

            [2] => Array
                (
                    [timestamp] => 1666888141
                    [category] => auth
                    [action] => Authenticated with * Using API Key
                    [metadata] => Array
                        (
                            [type] => info
                        )

                )

            [3] => Array
                (
                    [timestamp] => 1666888191
                    [category] => database
                    [action] => Found collection abcd1234
                    [metadata] => Array
                        (
                            [type] => info
                        )

                )

            [4] => Array
                (
                    [timestamp] => 1666888291
                    [category] => database
                    [action] => Permission for collection abcd1234 met
                    [metadata] => Array
                        (
                            [type] => debug
                        )

                )

            [5] => Array
                (
                    [timestamp] => 1666888491
                    [category] => database
                    [action] => Missing document when searching by ID!
                    [metadata] => Array
                        (
                            [type] => error
                        )

                )

        )

    [tags] => Array
        (
            [sdk] => Flutter
            [sdkVersion] => 0.0.1
            [authMode] => default
            [authMethod] => cookie
            [authProvider] => MagicLink
        )

    [params] => Array
        (
            [urgent] => false
            [isExpected] => true
            [file] => '/User/example/server/src/server/server.js'
            [line] => '15'
        )

)
```

Data in New Relic:
image

### CODE REVIEW

**Comment by ChandanChainani** [comment]
@boydbloemsma Please can you add `Closes #4209` to the description so the PR gets linked to the issue and after merging it closes the issue, it will also help others to know that there is a PR raised for the issue

**Comment by boydbloemsma** [comment]
> @boydbloemsma Please can you add `Closes #4209` to the description so the PR gets linked to the issue and after merging it closes the issue, it will also help others to know that there is a PR raised for the issue

Thanks for letting me know, updated!

**Comment by ChandanChainani** [comment]
Sorry @boydbloemsma since the issue is mentioned in another repository it is not being properly linked to fix it update it with this `[appwrite/appwrite#4213](https://github.com/appwrite/appwrite/issues/4209)` it would create a blue link which can be seen in other PRs.

File: src/Logger/Adapter/NewRelic.php [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+            'namespace' => $log->getNamespace(),
+            'server' => $log->getServer(),
+            'user' => [
+                'id' => $log->getUser()->getId(),
+                'email' => $log->getUser()->getEmail(),
+                'username' => $log->getUser()->getUsername(),
+            ],
+            'breadcrumbs' => $breadcrumbsArray,
+            'tags' => $tags,
+            'params' => $params,
```
Comment (stnguyen90) [inline src/Logger/Adapter/NewRelic.php:82]: how are `params` handled in new relic? I couldn't find this in the docs.

Comment (stnguyen90) [reply src/Logger/Adapter/NewRelic.php:82]: Should this be here if it's processed down below?

**Review by stnguyen90 — CHANGES_REQUESTED** [review]
Great PR! 🤯 We left some comments during the review, please check them out.

Could you please add screenshots of:

1. the data sent from the this adapter
2. what it looks like in new relic

File: src/Logger/Adapter/NewRelic.php (hunk 1 above)
Comment (boydbloemsma) [reply src/Logger/Adapter/NewRelic.php:82]: I've updated the description, there you can see how `params` are handled.
What exactly do you mean with it being processed down below?

**Comment by Meldiron** [comment]
Thank you so much for the PR 🤩. We're adding the `hacktoberfest-accepted` label to ensure this PR counts towards your Hacktoberfest contributions count. With that said, please stay active on this PR to address any comments once you receive a review. Happy Hacktoberfest! 🎃

File: src/Logger/Adapter/NewRelic.php (hunk 1 above)
Comment (stnguyen90) [reply src/Logger/Adapter/NewRelic.php:82]: > What exactly do you mean with it being processed down below?

Never mind 😅

File: src/Logger/Adapter/NewRelic.php [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+{
+    /**
+     * @var string (required, can be found in NewRelic -> Account dropdown -> API keys)
+     */
+    protected string $licenseKey;
+
+    /**
+     * @var string (optional, the api url where New Relic is reachable, in case of an EU New Relic account could
+     *              is 'https://log-api.eu.newrelic.com/log/v1'. defaults to 'https://log-api.newrelic.com/log/v1')
+     */
```
Comment (stnguyen90) [inline src/Logger/Adapter/NewRelic.php:22]: thanks for the helpful comments!

**Review by stnguyen90 — CHANGES_REQUESTED** [review]
Great PR! 🤯 We left some comments during the review, please check them out.

**Pull request closed without merging**